framework/test/unit/node/consensus/consensus.spec.ts

```diff
@@ -172,6 +179,42 @@ describe('consensus', () => {
 			expect(stateMachine.executeGenesisBlock).not.toHaveBeenCalled();
 			expect(chain.loadLastBlocks).toHaveBeenCalledTimes(1);
 		});
+
+		it('should not execute genesis block if stateRoot is invalid', async () => {
```

Comment:
```suggestion
		it('should fail initialization if stateRoot is invalid', async () => {
```
state root is checked after genesis block is being executed. same fore below